ime/app/src/main/java/com/anysoftkeyboard/AnySoftKeyboard.java

```diff
@@ -684,6 +716,8 @@ private void onNonFunctionKey(
                     } else {
                         handleCharacter(primaryCode, key, multiTapIndex, nearByKeyCodes);
                     }
+                } else {
+                    handleCharacter(primaryCode, key, multiTapIndex, nearByKeyCodes);
```

Comment:
Why do we need this call now? handleCharacter was not called before

Reply:
First call handles character when CTRL is pressed but character is out of range e.g. less than 32.
Second calls handles character when CTRL isn't pressed.